### PR TITLE
Automatically generate mappings and periodics for multi-arch supplemental images

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/test-platform 

Follow-ups:
- Presubmit to check the image generation
- Entry in Makefile

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
https://github.com/openshift/release/pull/33369 to fix the pylint job.

File: hack/image-mirroring/supplemental_ci_images_mirror_gen.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                        continue
+                    # Ignore commented lines or if the mapping source is registry.ci.openshift.org
+                    # to avoid mirroring from app.ci to an external source
+                    if line.startswith("#") or line.startswith(APPCI_REGISTRY):
+                        continue
+
+                    source, dest = line.split()
+
+                    if not dest.startswith(APPCI_REGISTRY):
+                        print("BUG: where are we mirroring to?")
```
Comment (jmguzik) [inline hack/image-mirroring/supplemental_ci_images_mirror_gen.py:102]: Could this comment be a little more descriptive?

File: hack/image-mirroring/supplemental_ci_images_mirror_gen.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import glob
+import shutil
+from ruamel.yaml import YAML
+
+APPCI_REGISTRY = "registry.ci.openshift.org"
+MAPPING_FILE_PREFIX = "mapping_"
+DIR_PATH = "core-services/image-mirroring"
+PERIODICS_FILE = "ci-operator/jobs/infra-image-mirroring-multi-arch.yaml"
+ARCHITECTURES = {
+    "arm64": "linux/arm64",
```
Comment (jmguzik) [inline hack/image-mirroring/supplemental_ci_images_mirror_gen.py:12]: Architecture listed here is only arm64, should the file name reflect that's only for aarch64 or alt archs?

File: ci-operator/jobs/infra-image-mirroring-multi-arch.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    workdir: true
+  labels:
+    ci.openshift.io/area: supplemental-ci-images-arm64
+    ci.openshift.io/role: infra
+  name: periodic-image-mirroring-supplemental-ci-images-arm64
+  spec:
+    containers:
+    - args:
+      - --skip-missing=true
+      - --filter-by-os=linux/arm64
```
Comment (jmguzik) [inline ci-operator/jobs/infra-image-mirroring-multi-arch.yaml:20]: name multi-arch but only arm64 in the args

**Review by jmguzik — CHANGES_REQUESTED** [review]

File: hack/image-mirroring/supplemental_ci_images_mirror_gen.py (hunk 2 above)
Comment (droslean) [reply hack/image-mirroring/supplemental_ci_images_mirror_gen.py:12]: Once we have more clusters that support different architectures, we should include them here as well.

File: ci-operator/jobs/infra-image-mirroring-multi-arch.yaml (hunk 3 above)
Comment (droslean) [reply ci-operator/jobs/infra-image-mirroring-multi-arch.yaml:20]: We currently only have arm64 support. Once we add the new clusters, there will be more jobs in this file for different architectures.

File: hack/image-mirroring/supplemental_ci_images_mirror_gen.py (hunk 1 above)
Comment (droslean) [reply hack/image-mirroring/supplemental_ci_images_mirror_gen.py:106]: I added a comment!

**Review by jmguzik — APPROVED** [review]

**Pull request merged**